flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/AccessExecutionVertex.java

```diff
@@ -85,4 +86,11 @@
 	 */
 	@Nullable
 	AccessExecution getPriorExecutionAttempt(int attemptNumber);
+
+	/**
+	 * Returns prior executions for this execution vertex.
+	 *
+	 * @return prior executions for this execution vertex
+	 */
+	EvictingBoundedList<ArchivedExecution> getPriorExecutionAttempts();
```

Comment:
I think it's not ideal to expose the mutable backing `EvictingBoundedList` from `AccessExecutionVertex`, which aims at being read-only. We can copy the items in the `EvictingBoundedList` into a `List<ArchivedExecution>` and make the result immutable.